api/models.py

```diff
@@ -1,15 +1,26 @@
 from django.db import models
 
 
-class Entity(models.Model):
+class BaseClass(models.Model):
     name = models.CharField(max_length=30)
-    location = models.CharField(max_length=120)
-    user = models.ForeignKey('auth.User')
 
     def __str__(self):
         return self.name
 
 
+class Entity(BaseClass):
+    location = models.CharField(max_length=120)
+    user = models.ForeignKey('auth.User')
+
+
+class Species(BaseClass):
+    pass
+
+
+class Breed(BaseClass):
+    species_field = models.ForeignKey(Species, on_delete=models.CASCADE)
```

Comment:
The `_field` word is to prevent [field clashes](https://stackoverflow.com/questions/34086729/django-1-9-field-clashes-with-the-field-of-non-existing-field-in-parent-model). Same for `breed_field` in `Animal`